specification/trafficmanager/resource-manager/Microsoft.Network/stable/2020-12-01/trafficmanager.json

```diff
@@ -0,0 +1,1659 @@
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group containing the Traffic Manager endpoint."
+          },
+          {
+            "name": "profileName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Traffic Manager profile."
+          },
+          {
+            "name": "endpointType",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The type of the Traffic Manager endpoint."
+          },
+          {
+            "name": "endpointName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Traffic Manager endpoint."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Traffic Manager endpoint.",
+            "schema": {
+              "$ref": "#/definitions/Endpoint"
+            }
+          },
+          "default": {
+            "description": "Default response. It will be deserialized as per the Error definition.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Endpoint-GET-External-WithGeoMapping": {
+            "$ref": "./examples/Endpoint-GET-External-WithGeoMapping.json"
+          },
+          "Endpoint-GET-External-WithLocation": {
+            "$ref": "./examples/Endpoint-GET-External-WithLocation.json"
+          },
+          "Endpoint-GET-External-WithSubnetMapping": {
+            "$ref": "./examples/Endpoint-GET-External-WithSubnetMapping.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Endpoints"
+        ],
+        "operationId": "Endpoints_CreateOrUpdate",
+        "description": "Create or update a Traffic Manager endpoint.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group containing the Traffic Manager endpoint to be created or updated."
+          },
+          {
+            "name": "profileName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Traffic Manager profile."
+          },
+          {
+            "name": "endpointType",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The type of the Traffic Manager endpoint to be created or updated."
+          },
+          {
+            "name": "endpointName",
```

Comment:
When Create, is endpointName also required in path?

Reply:
Correct, a distinct endpoint name must be provided.